tests/helpers/in-memory-ipfs.ts

```diff
@@ -1,8 +1,23 @@
-import Ipfs, { IPFS } from "ipfs-core"
-import Repo from "ipfs-repo"
-import { MemoryDatastore } from "interface-datastore"
+import * as fs from "fs"
+import * as dagPB from "@ipld/dag-pb"
+import * as Ipfs from "ipfs-core"
+
+import { createRepo } from "ipfs-repo"
+import { BlockCodec } from "multiformats/codecs/interface"
+import { MemoryDatastore } from "datastore-core/memory"
+import { MemoryBlockstore } from "blockstore-core/memory"
+
+import tempDir from "ipfs-utils/src/temp-dir.js"
+import type { IPFS } from "ipfs-core"
+
 
 export async function createInMemoryIPFS(): Promise<IPFS> {
+  const dir = tempDir()
+  fs.mkdirSync(dir)
+
+  const memoryDs = new MemoryDatastore()
+  const memoryBs = new MemoryBlockstore()
```

Comment:
@matheus23 Bit of double effort here, but needed to replace this to get the tests working with the new `ipfs-core`.

Reply:
Ah yeah damnit. Should've told you I've battled with that in the wnfs2 branch. There I had to update ipfs-core to work with the newer IPLD libs. In case you hit something else while updating ipfs-core, take a look at that branch. There might be something useful :)